src/Microsoft.AspNetCore.Razor.Language/Components/ComponentDocumentClassifierPass.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Razor.Language.Components
+{
+    internal class ComponentDocumentClassifierPass : DocumentClassifierPassBase
+    {
+        public static readonly string ComponentDocumentKind = "component.1.0";
+        private static readonly object BuildRenderTreeBaseCallAnnotation = new object();
+        private static readonly char[] PathSeparators = new char[] { '/', '\\' };
+        private static readonly char[] NamespaceSeparators = new char[] { '.' };
+
+        protected override string DocumentKind => ComponentDocumentKind;
+
+        protected override bool IsMatch(RazorCodeDocument codeDocument, DocumentIntermediateNode documentNode) => true;
```

Comment:
Same behavior as `MvcViewDocumentClassifierPass` - always returns true if the configuration is selected